### DISCUSSION TITLE
HTTP interface should be an independent service

### DISCUSSION THREAD

**Discussion by adeshpandey** [jookies/jasmin#1073] [Ideas]
I see we have coded http interface as highly coupled, component of Jasmin but I propose to have a separate http service/sub-project which can be scaled separately. 

I would propose to use FastAPI for the HTTP interface and make it a separate service in docker-compose setup. I'd love to contribute on this front.

**Comment by farirat** [comment]
hello @adeshpandey 
True, http api is highly coupled and a redesign is needed, i've tried to dedicate some time for it (and for decoupling smpp server api/clients as well) but unfortunately this is not possible for the time being.
Your contrib is welcome, i can assist you taking your way into the process and help taking the decisions.
